src/crd/model/mod.rs

```diff
@@ -1,65 +1,57 @@
 pub mod model;
 pub use model::*;
 
-use kube::{
-    api::{Api, Informer, PatchParams, WatchEvent},
-    client::APIClient,
-};
-use log::{error, info};
+use kube::{api::PatchParams, client::APIClient};
+use log::error;
 use serde_json::json;
 
+use crate::crd::gordo::{load_gordo_resource, GordoStatus};
 use crate::GordoEnvironmentConfig;
+use kube::api::Reflector;
 
 pub async fn monitor_models(client: &APIClient, namespace: &str, _env_config: &GordoEnvironmentConfig) -> ! {
-    let model_resource: Api<Model> = Api::customResource(client.clone(), "models")
-        .version("v1")
-        .group("equinor.com")
-        .within(&namespace);
-    let model_informer: Informer<Model> = Informer::new(model_resource.clone()).init().await.unwrap();
+    let model_resource = load_model_resource(&client, &namespace);
+    let gordo_resource = load_gordo_resource(&client, &namespace);
+
+    let model_reflector = Reflector::new(model_resource.clone()).init().await.unwrap();
+    let gordo_reflector = Reflector::new(gordo_resource.clone()).init().await.unwrap();
 
-    let mut outdated_version = false;
     loop {
-        // updates to models
-        model_informer
-            .poll()
-            .await
-            .unwrap_or_else(|e| panic!("Failed to poll model informer: {:?}", e));
+        let models = model_reflector.read().unwrap();
+        let gordos = gordo_reflector.read().unwrap();
 
-        while let Some(event) = model_informer.pop() {
-            if let Err(err) = handle_model_event(event, &model_resource).await {
-                error!("Watch event error for model informer: {:?}", err);
-                outdated_version = true;
-            };
-        }
+        // Compare each Gordo's n-models-built against the total models currently found for that Gordo
+        for gordo in gordos {
+            let n_models_built = models
+                .iter()
+                .filter(|model| {
+                    model
+                        .metadata
+                        .ownerReferences
+                        .iter()
+                        .any(|owner_ref| owner_ref.name == gordo.metadata.name)
```

Comment:
I think we should have a *and* project_version is the same

Reply:
It doesn't seem clear to me how project version gets associated with `Model`, at least not in the `ownerReferences` are you suggesting to compare the Model's `metadata.labels.label-for-project-version`?

How hard should we fail if labels change/aren't there?

Reply:
Its ok to use the labels as well.